src/Functions/EntropyLeardnedHash.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+// Reference to the paper about entropy-learned hashing:
+// https://bhentsch.github.io/doc/EntropyLearnedHashing.pdf
+
+#include <cstdint>
+#include <limits>
+#include <map>
+#include <string>
+#include <vector>
+
+// Key is a byte string
+namespace EntropyLearnedHashing
+{
+using Key = std::string;
+
+[[nodiscard]] Key getPartialKey(const std::string & key, const std::vector<size_t> & positions);
+
+class IDsManager
+{
+private:
+    static IDsManager * instance_ptr;
+    IDsManager() { }
```

Comment:
I am pretty sure you don't need the ctors on l. 23-25, i.e. the compiler-generated ones will work just fine.